### PR TITLE
[Web UI] Silencing entry dialog

### LINKED ISSUE #1287: [Web UI] Events Page: Silence
## 🕸 Related

Sketch:   https://github.com/sensu/sensu-go/issues/813

### PR DESCRIPTION
**Pull request by 10xjs**
## What is this change?

Add a modal dialog component for creating and editing silencing entries.

image

## Why is this change necessary?

closes #1287

## Does your change need a Changelog entry?

Nah.

## Do you need clarification on anything?

Nah.

## Were there any complications while making this change?

I created [10xjs/form](https://github.com/10xjs/form) and [10xjs/date-input-controller](https://github.com/10xjs/date-input-controller) with many tests to drive this and upcoming features. Less code in our repo is better, right? 😛

### CODE REVIEW

File: dashboard/config/webpack.config.prod.js [hunk 1]
```diff
@@ -144,7 +147,9 @@ module.exports = {
           // Process JS with Babel.
           {
             test: /\.(js|jsx)$/,
-            include: paths.appSrc,
+            include(file) {
+              return file.indexOf(paths.appSrc) === 0 || packageHasESNext(file);
+            },
```
Comment (jamesdphillips) [inline dashboard/config/webpack.config.prod.js]: This is great.

Can we move these changes into a separate PR? It's seemingly tangential to the changes and with squash and merge we will partially lose context.

Comment (10xjs) [reply dashboard/config/webpack.config.prod.js]: Absolutely. Good call.

**Comment by 10xjs** [comment]
@jamesdphillips Updated with much cleaner commits.

File: dashboard/src/components/util/RenderCounter.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+
+class RenderCounter extends React.Component {
```
Comment (jamesdphillips) [inline dashboard/src/components/util/RenderCounter.js]: is this being used?

Comment (10xjs) [reply dashboard/src/components/util/RenderCounter.js]: Not currently, but I want to keep it around as a debugging tool.

Comment (10xjs) [reply dashboard/src/components/util/RenderCounter.js]: It will print a running count of component updates and is useful for confirming that unnecessary renders aren't getting triggered as we work on different components.
